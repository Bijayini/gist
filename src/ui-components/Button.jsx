/* eslint-disable react/button-has-type */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import theme from '../lib/theme';

const Button = ({
  children,
  type,
  size,
  variant,
  handleClick,
  styles,
  className,
}) => {
  const getClassname = () =>
    `button ${size}-button ${variant}-button ${className} `;
  return (
    <Fragment>
      <button type={type} onClick={handleClick} className={getClassname()}>
        {children}
      </button>
      <style jsx="true">
        {`
          .button {
            font-family: Lato, Sans-Serif;
            font-weight: normal;
            font-size: 1.125rem;
            outline: none;
            border-radius: 3px;
            text-align: center;
            border: 0 none;
            cursor: pointer;
          }

          .primary-button {
            background-color: ${theme.primaryColor};
            color: ${theme.white};
            border: 1px solid ${theme.primaryColor};
          }

          .secondary-button {
            background-color: ${theme.white};
            color: ${theme.primaryColor};
            border: 1px solid ${theme.primaryColor};
          }

          .large-button {
            width: 100%;
            max-width: 300px;
            height: 42px;
            display: block;
            margin: 0 auto;
          }

          .small-button {
            font-size: 20px;
            padding: 6px 18px;
          }
        `}
      </style>
      {styles}
    </Fragment>
  );
};

Button.defaultProps = {
  type: 'button',
  variant: 'primary',
  className: '',
  styles: false,
};

Button.propTypes = {
  variant: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
  type: PropTypes.string,
  size: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  styles: PropTypes.node,
  className: PropTypes.string,
};

export default Button;
