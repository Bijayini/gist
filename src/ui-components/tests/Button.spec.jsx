import React from 'react';
import { shallow } from 'enzyme';

import Button from '../Button';

describe('<Button />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      variant: 'secondary',
      size: 'small',
      handleClick: jest.fn(),
    };
    component = shallow(<Button {...props}> Hello </Button>);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
