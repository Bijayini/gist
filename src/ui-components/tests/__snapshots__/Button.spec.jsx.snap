// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match with snapshot 1`] = `
"<Fragment>
  <button type=\\"button\\" onClick={[Function: mockConstructor]} className=\\"jsx-4236284080 button small-button secondary-button  \\">
     Hello 
  </button>
  <JSXStyle id=\\"395053386\\" dynamic={{...}}>
    .button.__jsx-style-dynamic-selector{font-family:Lato,Sans-Serif;font-weight:normal;font-size:1.125rem;outline:none;border-radius:3px;text-align:center;border:0 none;cursor:pointer;}.primary-button.__jsx-style-dynamic-selector{background-color:#cc1c5d;color:#ffffff;border:1px solid #cc1c5d;}.secondary-button.__jsx-style-dynamic-selector{background-color:#ffffff;color:#cc1c5d;border:1px solid #cc1c5d;}.large-button.__jsx-style-dynamic-selector{width:100%;max-width:300px;height:42px;display:block;margin:0 auto;}.small-button.__jsx-style-dynamic-selector{font-size:20px;padding:6px 18px;}
  </JSXStyle>
</Fragment>"
`;
